enigma-js/test/Enigma.spec.js

```diff
@@ -2221,5 +2233,176 @@ describe('Enigma tests', () => {
+      })).rejects.toEqual('Returned error: VM Exception while processing transaction: revert Not updated Enigma' +
+        ' contract');
+    });
+
+    it('should fail to commit receipt to old Enigma contract', async () => {
+      const gasUsed = 25;
+      const optionalEthereumData = '0x';
+      const optionalEthereumContractAddress = '0x0000000000000000000000000000000000000000';
+      stateDeltaHash = '0x0000000000000000000000000000000000000000000000000000000000000000';
+      const proof = utils.hash([
+        codeHash, pendingTaskA.inputsHash, initStateDeltaHash, stateDeltaHash, outputHash,
+        JSBI.BigInt(gasUsed).toString(16).padStart(16, '0'), optionalEthereumData, optionalEthereumContractAddress,
+        '0x01']);
+      const workerParams = await enigma.getWorkerParams(pendingTaskA.creationBlockNumber);
+      const selectedWorkerAddr = (await enigma.selectWorkerGroup(pendingTaskA.scAddr, workerParams, 1))[0];
+      let worker = await enigma.admin.findBySigningAddress(selectedWorkerAddr);
+      const priv = data.workers.find((w) => w[0] === selectedWorkerAddr.toLowerCase())[4];
+      const sig = EthCrypto.sign(priv, proof);
+      await expect(new Promise((resolve, reject) => {
+        enigma.enigmaContract.methods.commitReceipt(scAddr, pendingTaskA.taskId, stateDeltaHash, outputHash,
+          optionalEthereumData,
+          optionalEthereumContractAddress, gasUsed, sig).send({
+          from: worker.account,
+        }).on('receipt', (receipt) => resolve(receipt)).on('error', (error) => reject(error.message));
+      })).rejects.toEqual('Returned error: VM Exception while processing transaction: revert Not updated Enigma' +
+        ' contract');
+    });
+
+    it('should re-register workers and confirm worker balance has been transferred', async () => {
+      const workerParams = await enigma.getWorkerParams(latestTask.creationBlockNumber);
+      const selectedWorkerAddr = (await enigma.selectWorkerGroup(latestTask.scAddr, workerParams, 1))[0];
+      let worker = await enigma.admin.findBySigningAddress(selectedWorkerAddr);
+      const index = accounts.indexOf(worker.account);
+      const workerData = data.workers[index];
+      const report = '0x' + Array.from(workerData[1]).map((c) => c.charCodeAt(0).toString(16)).join('');
+      const signature = '0x' + workerData[3];
+      const proof = utils.hash([enigmaUpgradedContract.options.address]);
+      const priv = workerData[4];
+      const sig = EthCrypto.sign(priv, proof);
+      const startingOldContractBalance = parseInt(
+        await enigma.tokenContract.methods.balanceOf(enigma.enigmaContract.options.address).call(),
+      );
+      const startingNewContractBalance = parseInt(
+        await enigma.tokenContract.methods.balanceOf(enigmaUpgradedContract.options.address).call(),
+      );
+      await new Promise((resolve, reject) => {
+        enigmaUpgradedContract.methods.register(workerData[0], report, signature, sig).send({
```

Comment:
I'm confused because I don't see `transferWorkerStakePostUpgrade()` being called in the unit tests after the new contract has been upgraded. What am I missing?

Reply:
This is called inside the new contract's register function